src/test/projectsCreate.spec.ts

```diff
@@ -0,0 +1,220 @@
+          parentResource: PARENT_RESOURCE,
+        });
+      } catch (e) {
+        err = e;
+      }
+      expect(err.message).to.equal(
+        "Failed to create Google Cloud project. See firebase-debug.log for more info."
+      );
+      expect(err.original).to.equal(expectedError);
+      expect(apiRequestStub.callCount).to.equal(1);
+      expect(pollOperationStub.callCount).to.equal(1);
+    });
+
+    it("should reject if add Firebase api call fails", async () => {
+      const expectedError = new Error("HTTP Error 404: Not Found");
+      expectedCalledStubs.push(
+        _createCloudProjectApiStub().resolves({ body: { name: OPERATION_RESOURCE_NAME_1 } }),
+        _pollCreateCloudProjectOperationStub(OPERATION_RESOURCE_NAME_1).resolves({
+          projectNumber: PROJECT_NUMBER,
+        }),
+        _addFirebaseApiStub().rejects(expectedError)
+      );
+      mockOraWrapper.expects("start").exactly(2);
+      mockOraWrapper.expects("succeed").exactly(1);
+      mockOraWrapper.expects("fail").exactly(1);
+
+      let err;
+      try {
+        await createFirebaseProject(PROJECT_ID, {
+          displayName: PROJECT_NAME,
+          parentResource: PARENT_RESOURCE,
+        });
+      } catch (e) {
+        err = e;
+      }
+      expect(err.message).to.equal(
+        "Failed to add Firebase to Google Cloud Platform project. See firebase-debug.log for more info."
+      );
+      expect(err.original).to.equal(expectedError);
+      expect(apiRequestStub.callCount).to.equal(2);
+      expect(pollOperationStub.callCount).to.equal(1);
+    });
+
+    it("should reject if polling add Firebase operation returns error response", async () => {
+      const expectedError = new Error("Permission denied");
+      expectedCalledStubs.push(
+        _createCloudProjectApiStub().resolves({ body: { name: OPERATION_RESOURCE_NAME_1 } }),
+        _pollCreateCloudProjectOperationStub(OPERATION_RESOURCE_NAME_1).resolves({
+          projectNumber: PROJECT_NUMBER,
+        }),
+        _addFirebaseApiStub().resolves({ body: { name: OPERATION_RESOURCE_NAME_2 } }),
+        _pollAddFirebaseOperationStub(OPERATION_RESOURCE_NAME_2).rejects(expectedError)
+      );
+      mockOraWrapper.expects("start").exactly(2);
+      mockOraWrapper.expects("succeed").exactly(1);
+      mockOraWrapper.expects("fail").exactly(1);
+
+      let err;
+      try {
+        await createFirebaseProject(PROJECT_ID, {
+          displayName: PROJECT_NAME,
+          parentResource: PARENT_RESOURCE,
+        });
+      } catch (e) {
+        err = e;
+      }
+      expect(err.message).to.equal(
+        "Failed to add Firebase to Google Cloud Platform project. See firebase-debug.log for more info."
+      );
+      expect(err.original).to.equal(expectedError);
+      expect(apiRequestStub.callCount).to.equal(2);
+      expect(pollOperationStub.callCount).to.equal(2);
+    });
+
+    function _createCloudProjectApiStub(): sinon.SinonStub {
```

Comment:
Underscores generally denote private functions in JS files. Since this is a TS file and this function is not exported, that's enough. Please remove all the `_` prefixes

Reply:
Sure

Reply:
Done